rosbag2_transport/src/rosbag2_transport/qos.cpp

```diff
@@ -19,20 +19,37 @@
 
 #include "qos.hpp"
 
+namespace
+{
+// These values are returned to mean "unspecified" in FastDDS
+// 0 is equivalent and is portable to other implementation
```

Comment:
If `0` is equivalent, why do we need this change?
I feel generally we should document this in a more central way - also on the actual description of this PR for future references.

Reply:
Here's the full breakdown:

* Providing 0 as input duration to a publisher results 'unspecified' for all rmw implementations for all currently supported QoS policies
* FastDDS reports this special value (2147483647, 4294967295)  when you query a publisher that has an 'unspecified' duration 
  * FastDDS accepts this value as 'unspecified' when creating a publisher (the same as it does 0) - but it WON'T parse the CycloneDDS value
* CycloneDDS reports the special value (9223372036, 854775807) when you query a publisher that has an 'unspecified' duration
  * CycloneDDS accepts this value as 'unspecified' when creating a publisher (the same as it does 0) - but it WON'T parse the FastDDS value

The `rmw_cyclonedds_cpp` error looks like this when trying to read a bag recorded with `rmw_fastrtps_cpp`


```
2021-01-12T01:40:46.2371892Z 2: 1610415583.125080 [0] dq.builtin: invalid parameter list (vendor 1.16, version 2.1): pid 23 (DEADLINE) invalid, input = 3,0,0,128,6,250,130,75
2021-01-12T01:40:46.2372518Z 2: 1610415583.125090 [0] dq.builtin: SPDP (vendor 1.16): invalid qos/parameters
2021-01-12T01:40:46.2373131Z 2: 1610415583.126850 [0] dq.builtin: invalid parameter list (vendor 1.16, version 2.1): pid 23 (DEADLINE) invalid, input = 3,0,0,128,6,250,130,75
2021-01-12T01:40:46.2373762Z 2: 1610415583.126857 [0] dq.builtin: SPDP (vendor 1.16): invalid qos/parameters
```

The `rmw_fastrtps_cpp` error looks like this when trying to read a bag recorded with `rmw_cyclonedds_cpp`

```
[WARN] [1611257485.439465477] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
[WARN] [1611257485.439495434] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
[WARN] [1611257485.439502850] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
[WARN] [1611257485.440722168] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
[WARN] [1611257485.440737894] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
[WARN] [1611257485.440758600] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
[WARN] [1611257485.441663946] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
[WARN] [1611257485.441677598] [rmw_dds_common]: nanoseconds value too large for 32-bits, saturated at UINT_MAX
```

On this further investigation, it's actually worse than I thought - I think we need to definitively resolve  https://github.com/ros2/rmw/pull/255 in order to actually solve this problem.

FOR TODAY - I think the unblocking resolution would be to revert any special-case code here and just make the test-sample data use "0" across the board, which can be parsed and tested on any implementation

Reply:
awesome, thanks for the clarification.